Tests/AsyncHTTPClientTests/HTTPClientTestUtils.swift

```diff
@@ -49,11 +49,16 @@ func getDefaultEventLoopGroup(numberOfThreads: Int) -> EventLoopGroup {
 class TestHTTPDelegate: HTTPClientResponseDelegate {
     typealias Response = Void
 
-    init(backpressureEventLoop: EventLoop? = nil) {
+    init(
+        backpressureEventLoop: EventLoop? = nil,
+        stateDidChangeCallback: ((State) -> Void)? = nil
+    ) {
         self.backpressureEventLoop = backpressureEventLoop
+        self.stateDidChangeCallback = stateDidChangeCallback
     }
 
     var backpressureEventLoop: EventLoop?
+    var stateDidChangeCallback: ((State) -> Void)?
```

Comment:
this should be at least a let. however in your tests your mainly using this to get a callback once the head was received. I think, a specialized delegate only for this one job, might be a better idea.